pandas/io/parsers.py

```diff
@@ -1605,9 +1606,18 @@ def _cast_types(self, values, cast_type, column):
             # XXX this is for consistency with
             # c-parser which parses all categories
             # as strings
-            if not is_object_dtype(values):
-                values = astype_nansafe(values, str)
-            values = Categorical(values)
+            known_cats = (isinstance(cast_type, CategoricalDtype) and
+                          cast_type.categories is not None)
+
+            if known_cats:
+                cats = Index(values).unique()
+                values = Categorical._from_inferred_categories(
+                    cats, cats.get_indexer(values), cast_type
+                )
+            else:
```

Comment:
any reason you are not handling this case as well? (I get that it conflates the purpose of ``_from_inferred_categories`` a bit), but in reality this is just like passing ``dtype=None``.

I don't like to scatter casting/inferrence code around, very hard to figure out what's going on when when its not in 1 place.